internal/wasm/memory.go

```diff
@@ -73,18 +78,22 @@ func NewMemoryInstance(memSec *Memory) *MemoryInstance {
 		// the memory buffer allocation here is virtual and doesn't consume physical memory until it's used.
 		// 	* https://github.com/golang/go/blob/8121604559035734c9677d5281bbdac8b1c17a1e/src/runtime/malloc.go#L1059
 		//	* https://github.com/golang/go/blob/8121604559035734c9677d5281bbdac8b1c17a1e/src/runtime/malloc.go#L1165
-		size = MemoryPagesToBytesNum(memSec.Max)
-	} else {
-		size = MemoryPagesToBytesNum(memSec.Cap)
+		cur = max
 	}
 
-	buffer := make([]byte, MemoryPagesToBytesNum(memSec.Min), size)
+	var buffer []byte
+	if allocator != nil {
+		buffer = allocator.Make(min, cur, max)
+	} else {
+		buffer = make([]byte, min, cur)
+	}
```

Comment:
Opted for `nil` checks rather than a default implementation, because the zero value is used in a bunch of tests.